docs/reference/edot-collector/config/custom-data-streams.md

```diff
@@ -0,0 +1,118 @@
+---
+navigation_title: Custom data stream routing
+description: Customize data stream routing in EDOT. Learn scenarios, patterns, and risks when modifying data_stream.namespace or data_stream.dataset.
+applies_to:
+  stack:
+  serverless:
+    observability:
+  product:
+    edot_collector: ga
+products:
+  - id: cloud-serverless
+  - id: observability
+  - id: edot-collector
+---
+
+# Custom data stream routing with EDOT
+
+{{edot}} (EDOT) uses opinionated defaults for data stream naming to ensure compatibility with Elastic dashboards, {{product.apm}} visualizations, and curated UIs.
+
+While most use cases rely on these defaults, EDOT also supports advanced dynamic routing.
+
+:::{warning}
+We strongly recommend not changing the default data stream names. Customizing data stream routing diverges from the standard ingestion model and there's no guarantee it will be valid for future versions.
+:::
+
+## When to customize data streams
+
+The only recommended use case for customizing data stream routing is to separate data by environment (for example: dev, staging, and prod).
+
+In this case, we recommend changing only `data_stream.namespace`, not `data_stream.dataset`.
+
+## Modifying `namespace`
+
+A data stream name follows this structure:
+
+```
+<type>-<dataset>-<namespace>
+```
+
+The `namespace` is intended as the configurable part of the name. Elastic dashboards, detectors, and UIs support multiple namespaces automatically.
+
+## Why not modify `dataset`?
+
+Changing the `dataset` value can cause:
+
+- Dashboards and {{product.apm}} views to fail to load
+- Loss of compatibility with built-in correlations and cross-linking
+- Inconsistent field mappings
+- Proliferation of data streams and increased shard counts
+- Incompatibility with OpenTelemetry content packs, which are required to visualize OpenTelemetry data stored natively as OpenTelemetry semantic conventions
+
+Only modify `dataset` if it's absolutely necessary and you're aware of the tradeoffs.
+
+## Configuration example
+
+To enable dynamic data stream routing:
+
+1. Set `mapping.mode: otel` in the {{es}} exporter. When using `otel` mapping mode, the exporter appends `.otel` to the `data_stream.dataset` value automatically.
+2. Use a `resource` processor to set the desired `namespace` or `dataset` from resource attributes.
+3. Add the processor to your pipeline.
+
+:::{note}
+The example is purely illustrative, with no guarantee of it being production ready.
+:::
+
+```yaml
+exporters:
+  elasticsearch/otel:
+    api_key: ${env:ELASTIC_API_KEY}
+    endpoints: [${env:ELASTIC_ENDPOINT}]
+    mapping:
+      mode: otel
+
+processors:
+  resource/env-namespace: # <-- make sure you are using `resource` and not `attributes`
```

Comment:
<img width="418" height="111" alt="Image" src="https://github.com/user-attachments/assets/862edbcf-2cc1-43a1-b5e5-059a79566479" />

these arrows should be removed